### PR TITLE
Implement scry and sign with ethers.js

### PR DESCRIPTION
**Pull request by complected**
- I know you only asked me to implement `scry`. I was testing the APIs from ethersjs they seem to work well for `sign` as well. So I like to suggest ethersjs for implementing `sign` and remove `nacl` as dependency. If there is a reason you prefer `nacl`, please let me know.
- `scry`
- `sign`
- tests

### CODE REVIEW

**Comment by nmushegian** [comment]
Could you please make this use a crypto library directly instead of importing ethers, I think they are using this one 
https://www.npmjs.com/package/elliptic
https://github.com/ethers-io/ethers.js/blob/a71f51825571d1ea0fa997c1352d5b4d85643416/packages/signing-key/src.ts/index.ts#L79

File: coreword.js [hunk 1]
```diff
@@ -10,10 +10,19 @@ export function hash(b) {
...
 }
 
-export function sign(msg, key) {
-    return nacl.sign(msg, key)
+// Return EIP-191 signature for a UFT-8 encoded string and private key.
+// ***** CAUTION *****
+// https://docs.ethers.io/v5/api/signer/#Signer-signMessage
+// msg is converted to its representation in UTF-8 bytes before signing and understand its implications from the docs.
+// E.g. note that a string "0x1234" (6B) is **NOT** the same as [0x12, 0x34] (2B). 
+export async function sign(msg, key) {
```
Comment (nmushegian) [inline coreword.js:18]: These should both take UInt8Arrays or Buffers, we want to throw on strings to avoid confusion not require strings

**Comment by complected** [comment]
> Could you please make this use a crypto library directly instead of importing ethers, I think they are using this one ...

I looked into it. I have some 2nd thoughts and want to get you opinion first. 

Looks like what ethersJS may require a lot of/too much work to replace:

- For **verifying** and splitting the signature, looks like there is ethereum specific formatting in the signature, specifically the v in (r, s, v)

     - "In general, ECDSA signatures consist of two parameters, r and s. Signatures in Ethereum include a third parameter called v, that you can use to verify which account’s private key was used to sign the message..." [[Solidity]](https://docs.soliditylang.org/en/v0.8.14/solidity-by-example.html?highlight=ecrecover#recovering-the-message-signer-in-solidity)
    - the code has a few assumptions and domain knowledge e.g. "Allow a recid to be used as the v..." and "recoveryParam" [1](https://github.com/ethers-io/ethers.js/blob/a71f51825571d1ea0fa997c1352d5b4d85643416/packages/signing-key/src.ts/index.ts#L79-L80), [2](https://github.com/ethers-io/ethers.js/blob/a71f51825571d1ea0fa997c1352d5b4d85643416/packages/bytes/src.ts/index.ts#L333)
- For **signing**, there is prefixing / a small standard [[ethers]](https://docs.ethers.io/v5/api/signer/#Signer--signing-methods)
- I suspect there are a few relevant standards for verifying and signing to check

Just wondering if you think I should keep diving and if it's worth it, or if you want me to focus on namepet/minicash et al and use ethers JS, for now.

File: coreword.js [hunk 2]
```diff
@@ -10,10 +10,23 @@ export function hash(b) {
...
 }
 
-export function scry(msg, sig, key) {
-    return nacl.verify(msg, sig, key)
+// Return public key as is, only if the key is verified to have signed message. Throw an error otherwise.
+export function scry(msg, key, sig) {
+    if ( !(msg instanceof Uint8Array) || !(key instanceof Uint8Array)) {
+        throw new Error("Message and key must be Uint8Array.")
+    }
+    const addr = etils.verifyMessage(msg, sig)
```
Comment (nmushegian) [inline coreword.js:27]: This would be ok if we can grab the public key and not address

**Comment by nmushegian** [comment]
I’m ok with this if we can recover pubkey not address (pubkey hash), should be straightforward

**Comment by nmushegian** [comment]
The main purpose of this issue should be to recover the public key from signature, if we have the same interface then yeah we can just pick any lib and use it as is

File: test.js [hunk 3]
```diff
@@ -1,7 +1,33 @@
 import { test } from 'tapzero'
 
-import { roll } from './coreword.js'
+import { roll, scry, sign } from './coreword.js'
+
+import { utils as etils } from 'ethers'
```
Comment (nmushegian) [inline test.js:5]: `etils` that's a good one nice

**Comment by complected** [comment]
> I’m ok with this if we can recover pubkey not address (pubkey hash), should be straightforward

Right, we were actually returning the pubkey as is, after comparing the addresses. 

Now I return the same, but compare the pubkeys instead of addresses. I don't think comparing pubkeys is what you were asking for. Comparing pubkeys does have the benefit unnecessarily computing address [code](https://github.com/ethers-io/ethers.js/blob/a71f51825571d1ea0fa997c1352d5b4d85643416/packages/transactions/src.ts/index.ts#L115)

**Pull request merged**